src/ambianic/pipeline/store.py

```diff
@@ -174,16 +177,35 @@ def notify(self, save_json: dict):
         if self.notification is None:
             return
         log.debug("Sending notification(s)..")
+        directory = "{0}/ambianic-edge/premium.yaml".format(os.getcwd())
+
         # TODO extract inference data
         for inference_result in save_json['inference_result']:
-            data = {
-                'id': save_json['id'],
-                'label': inference_result['label'],
-                'confidence': inference_result['confidence'],
-                'datetime': save_json['datetime'],
-            }
-            notification = Notification(data=data, providers=self.notification_config["providers"])
```

Comment:
@vickywane a critical feature we need to consider for the notification service is potential failure to call the rest api due to internet disconnect or cloud service downtime. In such cases we need to have some kind of local queue for outgoing notifications up to size N and periodically retry to push them out.

Maybe this feature can be added to the edge notification service in general and it will benefit all notifications , not just premium.

Reply:
Can we use a queuing service for this? 

Or what are your recommendations?

Reply:
@vickywane this is still an open topic. Either address in this PR or open a separate issue.

Reply:
@vickywane this is still an open topic...

Reply:
[This issue](https://github.com/ambianic/ambianic-subscriptions/issues/12) has been opened in the `Ambianic-subscriptions` repository  to report the need for notification retries at intervals